### PR TITLE
feat(pkcs11-tool): testing HMAC

### PR DESCRIPTION
**Pull request by AlexandreGonzalo**
Hi All,

I wanted to test HMAC signature and verification with the new version of the pkc11-tool and I found several issues.
 - We need to add MF_GENERIC_HMAC_FLAGS for CKM_MD5_HMAC
 - We need --usage-sign for secret keys involved in HMAC operations
 - It would be great if we can add the CKM_SHA3_XXX_HMAC mechanisms definitions.

##### Checklist

- [ ] Documentation is added or updated
- [ ] New files have a LGPL 2.1 license statement
- [ ] PKCS#11 module is tested
- [ ] Windows minidriver is tested
- [ ] macOS tokend is tested

### CODE REVIEW

File: src/tools/pkcs11-tool.c [hunk 1]
```diff
@@ -7951,7 +7965,7 @@ static struct mech_info	p11_mechanisms[] = {
       { CKM_MD2_HMAC,		"MD2-HMAC", NULL, MF_UNKNOWN },
```
Comment (AlexandreGonzalo) [inline src/tools/pkcs11-tool.c:7965]: I can't test it but we also need MF_GENERIC_HMAC_FLAGS here

File: src/tools/pkcs11-tool.c [hunk 2]
```diff
@@ -7951,7 +7965,7 @@ static struct mech_info	p11_mechanisms[] = {
...
       { CKM_MD2_HMAC_GENERAL,	"MD2-HMAC-GENERAL", NULL, MF_UNKNOWN },
```
Comment (AlexandreGonzalo) [inline src/tools/pkcs11-tool.c:7966]: I can't test it but we also need MF_GENERIC_HMAC_FLAGS here

File: src/tools/pkcs11-tool.c [hunk 3]
```diff
@@ -7951,7 +7965,7 @@ static struct mech_info	p11_mechanisms[] = {
...
       { CKM_MD5,		"MD5", NULL, MF_UNKNOWN },
-      { CKM_MD5_HMAC,		"MD5-HMAC", NULL, MF_UNKNOWN },
+      { CKM_MD5_HMAC,		"MD5-HMAC", NULL, MF_GENERIC_HMAC_FLAGS },
       { CKM_MD5_HMAC_GENERAL,	"MD5-HMAC-GENERAL", NULL, MF_UNKNOWN },
```
Comment (AlexandreGonzalo) [inline src/tools/pkcs11-tool.c:7969]: I can't test it but we also need MF_GENERIC_HMAC_FLAGS here

File: src/tools/pkcs11-tool.c [hunk 4]
```diff
@@ -7965,9 +7979,13 @@ static struct mech_info	p11_mechanisms[] = {
...
+      { CKM_SHA3_224_HMAC,	"SHA3-224-HMAC", NULL, MF_GENERIC_HMAC_FLAGS },
       { CKM_SHA3_256,		"SHA3-256", NULL, MF_UNKNOWN },
+      { CKM_SHA3_256_HMAC,	"SHA3-256-HMAC", NULL, MF_GENERIC_HMAC_FLAGS },
       { CKM_SHA3_384,		"SHA3-384", NULL, MF_UNKNOWN },
+      { CKM_SHA3_384_HMAC,	"SHA3-384-HMAC", NULL, MF_GENERIC_HMAC_FLAGS },
       { CKM_SHA3_512,		"SHA3-512", NULL, MF_UNKNOWN },
+      { CKM_SHA3_512_HMAC,	"SHA3-512-HMAC", NULL, MF_GENERIC_HMAC_FLAGS },
       { CKM_RIPEMD128,		"RIPEMD128", NULL, MF_UNKNOWN },
       { CKM_RIPEMD128_HMAC,	"RIPEMD128-HMAC", NULL, MF_UNKNOWN },
       { CKM_RIPEMD128_HMAC_GENERAL,"RIPEMD128-HMAC-GENERAL", NULL, MF_UNKNOWN },
```
Comment (AlexandreGonzalo) [inline src/tools/pkcs11-tool.c:7991]: I can't test it but we also need MF_GENERIC_HMAC_FLAGS here

Comment (Jakuje) [reply src/tools/pkcs11-tool.c:7991]: please, add them also to any other mechanism that will require them even if you can not test it. I do not think the MD5 nor RIPEMD128 will be used somewhere in the future for anything useful, but if somebody will needed, they should not be blocked by the tool

Comment (AlexandreGonzalo) [reply src/tools/pkcs11-tool.c]: Please, have a look at my fourth commit.

Comment (Jakuje) [reply src/tools/pkcs11-tool.c]: looks good.

**Review by Jakuje — APPROVED** [review]

**Review by frankmorgner — APPROVED** [review]

**Comment by Jakuje** [comment]
Thank you for your contribution!

**Pull request merged**